### ISSUE TITLE
Updating docker image generates wallet loading error.

### ISSUE THREAD

**Issue by Yamaguchi** [chaintope/tapyrus-core#124]
### Describe the issue

1. Run docker image using docker-compose in [Tapyrus Explorer](https://github.com/chaintope/tapyrus-explorer)
    - ``` docker-compose up -d ```
    - [docker-compose.yml](https://github.com/chaintope/tapyrus-explorer/blob/bcc233452d9c6eda24d7cc2df032c5429ee97ad7/docker-compose.yml)

2. Create wallet and addresses
3. Stop docker container
    - ``` docker-compose down ```
4. Update docker image from `tapyrus/tapyrusd:latest` to `tapyrus/tapyrusd:edge`
    - tapyrus/tapyrusd:latest
    (DIGEST:sha256:764a269b5139328eaa5ff8c38e4ce7d290b5339812989f9bd8fb77bd500e099f)
    - tapyrus/tapyrusd:edge
    (DIGEST:sha256:81a14c82e44aae434465fcc61e48fa6fa46338780cebc389ed929b67be40caed)
5. Run again
    - ``` docker-compose up -d ```

### What behavior did you expect?

- Start tapyrusd process successfully.

### What was the actual behavior

- Failed to start. ` docker-compse logs tapyrusd ` shows the following:

```
...
tapyrusd_1  | 2021-03-23T08:42:38Z Using obfuscation key for /var/lib/tapyrus/prod-1939510133/chainstate: 4f808c2651502b59
tapyrusd_1  | 2021-03-23T08:42:38Z init message: Rewinding blocks...
tapyrusd_1  | 2021-03-23T08:42:38Z  block index               6ms
tapyrusd_1  | 2021-03-23T08:42:38Z ERROR: CBlockPolicyEstimator::Read(): up-version (1000000) fee estimate file
tapyrusd_1  | 2021-03-23T08:42:38Z init message: Loading wallet...
tapyrusd_1  | 2021-03-23T08:42:38Z Error: Error loading : Wallet requires newer version of Tapyrus Core
tapyrusd_1  | 2021-03-23T08:42:38Z [default wallet] Releasing wallet
tapyrusd_1  | Error: Error loading : Wallet requires newer version of Tapyrus Core
tapyrusd_1  | 2021-03-23T08:42:38Z Shutdown: In progress...
tapyrusd_1  | 2021-03-23T08:42:38Z scheduler thread interrupt
tapyrusd_1  | 2021-03-23T08:42:38Z Shutdown: done
tapyrus-explorer_tapyrusd_1 exited with code 1

```

**Comment by Yamaguchi** [comment]
Deleting wallet and recreating it worked for me

**Comment by Naviabheeman** [comment]
I am not able to simulate this error. I tried to crate a wallet in tapyrus4.0, copy the **wallet.dat** file into a new tapyrus4.1 container and start it. It is successful. 

Trying to change the version of Tapyrusd using locally built binaries didn't work as the container did not start when Tapyrus is changed.

2021-09-19T06:33:39Z Tapyrus Core version v0.4.1 (release build)
2021-09-19T06:33:39Z InitParameterInteraction: parameter interaction: -whitelistforcerelay=1 -> setting -whitelistrelay=1
2021-09-19T06:33:39Z Validating signatures for all blocks.
2021-09-19T06:33:39Z Using the 'sse4(1way),sse41(4way),avx2(8way)' SHA256 implementation
2021-09-19T06:33:39Z Using RdRand as an additional entropy source
2021-09-19T06:33:39Z Default data directory /root/.tapyrus
2021-09-19T06:33:39Z Using data directory /var/lib/tapyrus/dev-1905960821
2021-09-19T06:33:39Z Using config file /etc/tapyrus/tapyrus.conf
2021-09-19T06:33:39Z Using at most 125 automatic connections (1048576 file descriptors available)
2021-09-19T06:33:39Z Reading Genesis Block from [/var/lib/tapyrus/genesis.1905960821]
2021-09-19T06:33:39Z Genesis Block [d3769de4ba6bb5c50981d4b12a645a5028934ef0959b6a7915a3eb93029667a2] of Tapyrus network [1905960821] Loaded successfully
2021-09-19T06:33:39Z Using 16 MiB out of 32/2 requested for signature cache, able to store 524288 elements
2021-09-19T06:33:39Z Using 16 MiB out of 32/2 requested for script execution cache, able to store 524288 elements
2021-09-19T06:33:39Z Using 4 threads for script verification
2021-09-19T06:33:39Z scheduler thread start
2021-09-19T06:33:39Z libevent: getaddrinfo: address family for nodename not supported
2021-09-19T06:33:39Z Binding RPC on address :: port 12381 failed.
2021-09-19T06:33:39Z HTTP: creating work queue of depth 16
2021-09-19T06:33:39Z Config options rpcuser and rpcpassword will soon be deprecated. Locally-run instances may remove rpcuser to use cookie-based auth, or may be replaced with rpcauth. Please see share/rpcauth for rpcauth auth generation.
2021-09-19T06:33:39Z HTTP: starting 4 worker threads
2021-09-19T06:33:39Z Using wallet directory /var/lib/tapyrus/dev-1905960821/wallets
2021-09-19T06:33:39Z init message: Verifying wallet(s)...
2021-09-19T06:33:39Z Using BerkeleyDB version Berkeley DB 4.8.30: (April  9, 2010)
2021-09-19T06:33:39Z Using wallet wallet.dat
2021-09-19T06:33:39Z BerkeleyEnvironment::Open: LogDir=/var/lib/tapyrus/dev-1905960821/wallets/database ErrorFile=/var/lib/tapyrus/dev-1905960821/wallets/db.log
2021-09-19T06:33:39Z Cache configuration:
2021-09-19T06:33:39Z * Using 2.0MiB for block index database
2021-09-19T06:33:39Z * Using 8.0MiB for chain state database
2021-09-19T06:33:39Z * Using 440.0MiB for in-memory UTXO set (plus up to 286.1MiB of unused mempool space)
2021-09-19T06:33:39Z init message: Loading block index...
2021-09-19T06:33:39Z Opening LevelDB in /var/lib/tapyrus/dev-1905960821/blocks/index
2021-09-19T06:33:39Z Opened LevelDB successfully
2021-09-19T06:33:39Z Using obfuscation key for /var/lib/tapyrus/dev-1905960821/blocks/index: 0000000000000000
2021-09-19T06:33:39Z LoadBlockIndexDB: last block file = 0
2021-09-19T06:33:39Z LoadBlockIndexDB: last block file info: CBlockFileInfo(blocks=1, size=345, heights=0...0, time=2019-06-28...2019-06-28)
2021-09-19T06:33:39Z Checking all blk files are present...
2021-09-19T06:33:39Z Opening LevelDB in /var/lib/tapyrus/dev-1905960821/chainstate
2021-09-19T06:33:39Z Opened LevelDB successfully
2021-09-19T06:33:39Z Using obfuscation key for /var/lib/tapyrus/dev-1905960821/chainstate: 113abe5fc83e1dc3
2021-09-19T06:33:39Z Loaded best chain: hashBestChain=d3769de4ba6bb5c50981d4b12a645a5028934ef0959b6a7915a3eb93029667a2 height=0 date=2019-06-28T02:38:02Z progress=1.000000
2021-09-19T06:33:39Z init message: Rewinding blocks...
2021-09-19T06:33:39Z init message: Verifying blocks...
2021-09-19T06:33:39Z  block index              13ms
2021-09-19T06:33:39Z ERROR: CBlockPolicyEstimator::Read(): up-version (1000000) fee estimate file
2021-09-19T06:33:39Z init message: Loading wallet...
2021-09-19T06:33:39Z [default wallet] nFileVersion = 40000
2021-09-19T06:33:39Z [default wallet] Keys: 12 plaintext, 0 encrypted, 12 w/ metadata, 12 total. Unknown wallet records: 0
2021-09-19T06:33:39Z [default wallet] Wallet completed loading in               4ms
2021-09-19T06:33:39Z [default wallet] keypool added 1 keys (0 internal), size=2 (1 internal)
2021-09-19T06:33:39Z [default wallet] setKeyPool.size() = 2
2021-09-19T06:33:39Z [default wallet] mapWallet.size() = 8
2021-09-19T06:33:39Z [default wallet] mapAddressBook.size() = 2
2021-09-19T06:33:39Z mapBlockIndex.size() = 1
2021-09-19T06:33:39Z nBestHeight = 0
2021-09-19T06:33:39Z torcontrol thread start
2021-09-19T06:33:39Z Imported mempool transactions from disk: 0 succeeded, 0 failed, 0 expired, 0 already there
2021-09-19T06:33:39Z Bound to [::]:12383
2021-09-19T06:33:39Z Bound to 0.0.0.0:12383
2021-09-19T06:33:39Z init message: Loading P2P addresses...
2021-09-19T06:33:39Z Loaded 0 addresses from peers.dat  0ms
2021-09-19T06:33:39Z init message: Loading banlist...
2021-09-19T06:33:39Z init message: Starting network threads...
2021-09-19T06:33:39Z net thread start
2021-09-19T06:33:39Z addcon thread start
2021-09-19T06:33:39Z dnsseed thread start
2021-09-19T06:33:39Z Loading addresses from DNS seeds (could take a while)
2021-09-19T06:33:39Z init message: Done loading
2021-09-19T06:33:39Z opencon thread start
2021-09-19T06:33:39Z 0 addresses found from DNS seeds
2021-09-19T06:33:39Z dnsseed thread exit
2021-09-19T06:33:39Z msghand thread start

**Comment by Naviabheeman** [comment]
I was able to replicate this issue.

I created two containers with tapyrus v0.4.1 and tapyrus v0.5.1 and mounted a separate volume for walletdir. In each environment I created one wallet with 3 transactions. Then I swapped the wallet dir in both environments. And restarted the containers.  tapyrus v0.4.1 started successfully. tapyrus v0.5.1 failed to start. 

On examining further I found that v0.4.1 wallet has a wallet version more than v0.5.1
>% docker exec tapyrus_node_testnet_v41 tapyrus-cli -conf=/etc/tapyrus/tapyrus.conf getwalletinfo     
{
  "walletname": "",
  **"walletversion": 169900,**
  "balance": 100.00000000,
  "unconfirmed_balance": 0.00000000,
  "txcount": 3,
  "keypoololdest": 1659284530,
  "keypoolsize": 999,
  "keypoolsize_hd_internal": 1000,
  "paytxfee": 0.00000000,
  "hdseedid": "c15ea597fb612fe6e21b950ee09a0969a5b1254f",
  "hdmasterkeyid": "c15ea597fb612fe6e21b950ee09a0969a5b1254f",
  "private_keys_enabled": true
}

>% docker exec tapyrus_node_testnet_v5 tapyrus-cli -conf=/etc/tapyrus/tapyrus.conf getwalletinfo 
{
  "walletname": "",
  **"walletversion": 10500,**
  "balance": {
  },
  "unconfirmed_balance": {
  },
  "txcount": 3,
  "keypoololdest": 1659284595,
  "keypoolsize": 1000,
  "keypoolsize_hd_internal": 1000,
  "paytxfee": 0.00000000,
  "hdseedid": "02e0eb5ba95f5899307d5fe6d66394fe092ca776",
  "hdmasterkeyid": "02e0eb5ba95f5899307d5fe6d66394fe092ca776",
  "private_keys_enabled": true
}

**"walletversion": 169900,** was used before commit a61fb54. The new wallet version **"walletversion": 10500,** is used in all new environments.
I think this issue is not relevant now as the old wallet version will not be used anymore. 
We are close it unless some other specific test is needed.
